### DISCUSSION TITLE
Named instances for multiple existing keyvaults

### DISCUSSION THREAD

**Discussion by iamandymcinnes** [microsoft/aspire#5917] [General]
Hi,

I'm trying to connect to multiple keyvaults and wanted to use the named approach so I can resolve the relevant one at runtime.

The documentation [here](https://learn.microsoft.com/en-us/dotnet/aspire/security/azure-security-key-vault-integration?tabs=dotnet-cli#named-instances) seems to suggest that you can specify your config including the vaultURI and then just reference the name of that vault in the config section:

```
builder.AddAzureKeyVaultSecrets("INSTANCE_NAME");

{
  "Aspire": {
    "Azure": {
      "Security": {
        "KeyVault": {
          "INSTANCE_NAME": {
            "VaultUri": "YOUR_VAULT_URI",
            "DisableHealthChecks": false,
            "DisableTracing": true,
            "ClientOptions": {
              "DisableChallengeResourceVerification": true
            }
          }
        }
      }
    }
  }
}
```

However when you do that, aspire tries to generate a keyvault and complains that you haven't provide the subscription details etc.

Looking at the AddConnectionString() approach, there is no way to have named instances.

Is the documentation misleading or is there another way to achieve this?

Many thanks

Andy

**Comment by davidfowl** [comment]
cc @eerhardt

**Comment by eerhardt** [comment]
> However when you do that, aspire tries to generate a keyvault and complains that you haven't provide the subscription details etc.

Can you provide repro code for this? Also the exception message and callstack.

> Looking at the AddConnectionString() approach, there is no way to have named instances.

You put the connection string under the named section of `ConnectionStrings`. So:

```json
{
  "ConnectionStrings": {
    "INSTANCE_NAME": "YOUR_VAULT_URI"
    "INSTANCE_NAME2": "YOUR_VAULT_URI2"
  }
}
```
